### PR TITLE
Cherry pick of #94389: add lock for csi node update

### PR DESCRIPTION
**Pull request by ping035627**
Cherry pick of #94389 on release-1.19.

#94389: add lock for csi node update

/sig storage
/kind bug

### CODE REVIEW

**Comment by ping035627** [comment]
@vladimirvivien Hi, can you review this pr? Thanks!

File: pkg/volume/csi/nodeinfomanager/nodeinfomanager.go [hunk 1]
```diff
@@ -175,6 +178,9 @@ func (nim *nodeInfoManager) updateNode(updateFuncs ...nodeUpdateFunc) error {
 // the effects of previous updateFuncs to avoid potential conflicts. For example, if multiple
 // functions update the same field, updates in the last function are persisted.
 func (nim *nodeInfoManager) tryUpdateNode(updateFuncs ...nodeUpdateFunc) error {
+	nim.lock.Lock()
```
Comment (vladimirvivien) [inline pkg/volume/csi/nodeinfomanager/nodeinfomanager.go:181]: Has this been tested? Does it actually prevent race conditions when updating nodes info ?

**Pull request closed without merging**